doc/lspconfig.txt

```diff
@@ -1,242 +1,659 @@
+         filetypes = {'lua'};
+         root_dir = function(fname)
+           return lspconfig.util.find_git_ancestor(fname)
+         end;
+         settings = {};
+       };
+     }
+   end
+
+- call `setup()` to enable the FileType autocmd
+>
+   lspconfig.foo_lsp.setup{}
+<
 
-* `:LspInfo` shows the status of active and configured language servers.
-The following support tab-completion for all arguments:
-* `:LspStart <config_name>` Start the requested server name. Will only successfully start if the command detects a root directory matching the current config. Pass `autostart = false` to your `.setup{}` call for a language server if you would like to launch clients solely with this command. Defaults to all servers matching current buffer filetype.
-* `:LspStop <client_id>` Defaults to stopping all buffer clients.
-* `:LspRestart <client_id>` Defaults to restarting all buffer clients.
 ==============================================================================
-10. The wiki                                              *lspconfig-the-wiki*
+COMMANDS                                                    *lspconfig-commands*
+
+- `:LspInfo` shows the status of active and configured language servers. Note
+  that client id refers to the neovim RPC instance connected to a given
+  language server.
 
+The following commands support tab-completion for all arguments. All commands
+that require a client id can either leverage tab-completion or the info
+contained in `:LspInfo`:
+
+- `:LspStart <config_name>` Launch requested configured client. Attempts to
+  detect a root directory for the given client and fails if not found.
+  Defaults to all *configured* servers matching the current filetype.
+- `:LspStop <client_id>` Stop the server with the given client id. Defaults to
+  stopping all servers active on the current buffer.
+- `:LspRestart <client_id>` Restart the client with the given client id. Will
+  attempt to reattach to all previously attached buffers.
 
-Please see the [wiki](https://github.com/neovim/nvim-lspconfig/wiki) for
-additional topics, including:
-* [Installing language servers automatically](https://github.com/neovim/nvim-lspconfig/wiki/Installing-language-servers-automatically)
-* [Snippets support](https://github.com/neovim/nvim-lspconfig/wiki/Snippets)
-* [Project local settings](https://github.com/neovim/nvim-lspconfig/wiki/Project-local-settings)
-* [Recommended plugins for enhanced language server features](https://github.com/neovim/nvim-lspconfig/wiki/Language-specific-plugins)
 ==============================================================================
-11. Windows                                                *lspconfig-windows*
+ROOT DETECTION                                        *lspconfig-root-detection*
+                                                            *lspconfig-root-dir*
+
+lspconfig automatically launches language servers by defining a filetype
```

Comment:
Maybe an explanation/refresher first that this an LSP requirement
`:h vim.lsp.start_client()`
> Directory where the LSP server will base its rootUri on initialization.

Reply:
It's not required by the spec, it's assumed by many language servers.

In fact we send nil in #1385